src/lib/utils/matomo.ts

```diff
@@ -1,9 +1,30 @@
 import { push } from "@socialgouv/matomo-next"
 
+import { DEFAULT_LOCALE, LOCALES_CODES } from "@/lib/constants"
+
 import { IS_PROD } from "./env"
 
 export const MATOMO_LS_KEY = "ethereum-org.matomo-opt-out"
 
+/**
+ * Normalizes paths to ensure consistent Matomo tracking.
+ * With localePrefix: "as-needed", English paths don't have /en prefix,
+ * but we want to track them as /en paths for analytics consistency.
+ */
+export const normalizePathForMatomo = (pathname: string): string => {
+  // If already has a locale prefix (like /es/, /fr/), keep it as is
+  const hasLocalePrefix = LOCALES_CODES.some(
+    (locale) => locale !== DEFAULT_LOCALE && pathname.startsWith(`/${locale}/`)
+  )
+
+  if (hasLocalePrefix) {
+    return pathname
+  }
+
+  // For paths without locale prefix (English content), add /en prefix
+  return pathname === "/" ? "/en" : `/en${pathname}`
```

Comment:
@pettinarip In current production, when I log `pathname` for the homepage it prints `/en/` with the trailing slash (which is then logged to Matomo)... is there a reason we're forcing `/` => `/en` without the trailing slash here?

I think we can just do this:

```suggestion
  return `/${DEFAULT_LOCALE}${pathname}`
```

So `/` would return `/en/`, or `/some-path` would return `/en/some-path`